server/app/repository/ProgramRepository.java

```diff
@@ -123,6 +126,35 @@ public ImmutableSet<String> getAllProgramNames() {
     return names.build();
   }
 
+  /**
+   * Gets the program definition that contains the related question data from the cache (if
+   * enabled).
+   */
+  public Optional<ProgramDefinition> getFullProgramDefinitionFromCache(ProgramModel program) {
+    if (settingsManifest.getQuestionCacheEnabled()) {
+      return programDefCache.get(String.valueOf(program.id));
+    }
+    return Optional.empty();
+  }
+
+  public Optional<ProgramDefinition> getFullProgramDefinitionFromCache(long programId) {
+    if (settingsManifest.getQuestionCacheEnabled()) {
+      return programDefCache.get(String.valueOf(programId));
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Sets the program definition that contains the related question data in the cache (if enabled).
+   *
+   * <p>Draft program definition data must not be set in the cache.
+   */
+  public void setFullProgramDefinitionCache(long programId, ProgramDefinition programDefinition) {
```

Comment:
question (non-blocking): Are there any times we should remove a program definition from the cache?

Reply:
We only need to remove it from the cache in the DevDatabaseSeedController so that for the browser test runs the cache is cleared when the database is reset. In application code, we don't have to clear the cache because once a program is set to active or obsolete, the program data for that ID will never change.

Reply:
Got it, thanks!!